dali/test/python/test_operator_normal_distribution.py

```diff
@@ -14,109 +14,67 @@
 
-
-def check_normal_distribution_with_input(dtype, device):
-    input_data = [
-        [np.ones((1000)), np.ones((1000))],
-        [np.ones((100, 100)), np.ones((100, 100))],
-        [np.ones((100, 10, 10)), np.ones((100, 10, 10))]
-    ]
-    for batch in input_data:
-        bsize = len(batch)
-        pipeline = NormalDistributionPipelineWithInput(premade_batch=batch, dtype=dtype, device=device)
-        pipeline.build()
-        outputs = pipeline.run()
-        for i in range(bsize):
-            output = outputs[0] if device == 'cpu' else outputs[0].as_cpu()
-            assert output.at(i).shape == batch[i].shape
-            possibly_normal_distribution = output.at(i).flatten()
-            _, pvalues_anderson, _ = st.anderson(possibly_normal_distribution, dist='norm')
-            # It's not 100% mathematically correct, but makes do in case of this test
-            assert pvalues_anderson[2] > 0.5
-
-
-def test_normal_distribution_with_input():
-    for device in ['cpu', 'gpu']:
-        for t in test_types:
-            yield check_normal_distribution_with_input, t, device
-
-
-def check_normal_distribution_with_argument(shape, dtype, device):
-    pipeline = NormalDistributionPipelineWithArgument(shape, dtype=dtype, device=device)
-    pipeline.build()
-    outputs = pipeline.run()
-    output = outputs[0] if device == 'cpu' else outputs[0].as_cpu()
-    possibly_normal_distribution = output.as_array().flatten()
-    _, pvalues_anderson, _ = st.anderson(possibly_normal_distribution, dist='norm')
-    # It's not 100% mathematically correct, but makes do in case of this test
-    assert pvalues_anderson[2] > 0.5
-
-
-def test_normal_distribution_with_argument():
-    shapes = [[100], [10, 20, 30], [1, 2, 3, 4, 5, 6]]
-    for device in ['cpu', 'gpu']:
-        for shape in shapes:
-            for t in test_types:
-                yield check_normal_distribution_with_argument, shape, t, device
-
-
-def check_normal_distribution_default(dtype, device):
-    pipeline = NormalDistributionPipelineDefault(batch_size=100, dtype=dtype, device=device)
-    pipeline.build()
-    outputs = pipeline.run()
-    output = outputs[0] if device == 'cpu' else outputs[0].as_cpu()
-    possibly_normal_distribution = output.as_array().flatten()
-    _, pvalues_anderson, _ = st.anderson(possibly_normal_distribution, dist='norm')
-    # It's not 100% mathematically correct, but makes do in case of this test
-    assert pvalues_anderson[2] > 0.5
-
-
-def test_normal_distribution_default():
-    for device in ['cpu', 'gpu']:
-        for t in test_types:
-            yield check_normal_distribution_default, t, device
+def check_normal_distribution(device, dtype, shape=None, use_shape_like_input=False, mean=0.0, stddev=1.0,
+                              niter=3, batch_size=3, device_id=0, num_threads=3):
+    pipe = Pipeline(batch_size=batch_size, device_id=device_id, num_threads=num_threads, seed=123456)
+    with pipe:
+        if shape is not None:
```

Comment:
I'd use sth like this:
```Python
def random_shape(max_shape):
  return [np.random.randint(0, s) for s in max_shape]...

shape_like_in = None
shape_arg = None
if variable_shape:
  if shape_like:
    shape_like_in = fn.external_source(lambda: np.zeros(random_shape(shape)), batch=False)
  else:
    shape_arg = fn.external_source(lambda: random_shape(shape), batch=False)
else:
  if shape_like:
    shape_like_in = np.zeros(shape)
  else:
    shape_arg = shape
```
...and then use shape_like_in and shape_arg
```
inputs = [shape_like_in] if shape_like_in else []
fn.random.normal(*inputs, shape=shape_arg, device=device, mean=mean, stddev=stddev, dtype=dtype)
```
Also, use per-sample mean and stddev.